### PR TITLE
Enhance Virtual Machine and Device Platform Filter with Manufacturer Information

### LINKED ISSUE #12741: Add manufacturer to platform of virtual machines
### NetBox version

v3.5.2

### Feature type

Data model extension

### Proposed functionality

Add "manufacturer" field before "platform" selection inside virtual machine configuration

### Use case

Currently, when you create a device you should choose it's vendor and platform, and drop down list of platform filters to that vendor.
With that approach you can create platforms as software versions with specific manufacturer.

But when you choose platform for virtual machine it is one list which becomes confusing without vendor

### Database changes

Add field "manufacturer" to Virtualization -> Virtual Machine to filter platforms

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by kprince28**

### CODE REVIEW

File: netbox/virtualization/forms/filtersets.py [hunk 1]
```diff
@@ -154,11 +154,19 @@ class VirtualMachineFilterForm(
         choices=VirtualMachineStatusChoices,
         required=False
     )
+    manufacturer_id = DynamicModelMultipleChoiceField(
+        queryset=Manufacturer.objects.all(),
+        required=False,
+        label=_('Manufacturer'),
+    )
```
Comment (jeremystretch) [inline netbox/virtualization/forms/filtersets.py:161]: There is no `manufacturer_id` filter for virtual machines, so this form field would have no effect beyond constraining the choices for the platform field below, which could be very confusing to the user. The scope of FR #12741 deals only with the VM & device edit forms; we shouldn't be making any changes to the list filters as part of this work anyway.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/virtualization/forms/model_forms.py [hunk 2]
```diff
@@ -197,10 +197,21 @@ class VirtualMachineForm(TenancyForm, NetBoxModelForm):
...
         }
     )
+    manufacturer = DynamicModelChoiceField(
+        label=_('Manufacturer'),
+        queryset=Manufacturer.objects.all(),
+        required=False,
+        initial_params={
+            'platforms': '$platform'
+        }
+    )
```
[the remark this answers is not in the document]
Comment (kprince28) [reply netbox/virtualization/forms/model_forms.py:207]: @jeremystretch,
I'm curious about the `Advanced Selector Widget` and how it operates. Have you had the opportunity to implement it in your code? If so, I'd highly appreciate it if you could provide some references or perhaps share a few examples. Your expertise and insights would be immensely helpful in enhancing my understanding of this widget.

[the remark this answers is not in the document]
Comment (DanSheps) [reply netbox/virtualization/forms/model_forms.py:207]: You remove `initial_params={}` and instead use `selector=True`

**Review by DanSheps — APPROVED** [review]

**Comment by jeremystretch** [comment]
Thanks @kprince28!

**Pull request merged**